src/yarp.c

```diff
@@ -8304,7 +8304,12 @@ parse_expression(yp_parser_t *parser, yp_binding_power_t binding_power, const ch
   // Otherwise we'll look and see if the next token can be parsed as an infix
   // operator. If it can, then we'll parse it using parse_expression_infix.
   yp_binding_powers_t current_binding_powers;
-  while (current_binding_powers = yp_binding_powers[parser->current.type], binding_power <= current_binding_powers.left) {
+  while (
+    current_binding_powers = yp_binding_powers[parser->current.type],
+    binding_power <= current_binding_powers.left &&
+    current_binding_powers.right != YP_BINDING_POWER_UNARY &&
```

Comment:
I definitely agree this is the right solution but I'm nervous about this accidentally introducing something we're going to have to remember to change.

What do you think about introducing a `binary` attribute to the `binding_powers_t` struct instead? Then here we would check `&& current_binding_powers.binary` instead.